### DISCUSSION TITLE
Allow easier customisation of top and bottom tabs in shell

### DISCUSSION THREAD

**Discussion by AswinPG** [dotnet/maui#7167] [Ideas]
Shell is an awesome control but i could never really use it in it's full glory. Maui shell provides options for flyout, bottom tabs and top tabs. In the case of flyout, shell is doing a great job since we can use the control template to do whatever we want with the looks. But this is not the case for bottom and top tabs. The design freedom we developers have is strictly limited. This very problem is a major downside for using shell.
For me I never actually used shell because of this. 

Is there an easy way to swap the native tab rendering to cross platform tab rendering as suggested in this spec : 
https://github.com/xamarin/Xamarin.Forms/issues/10773 

To overcome this issue we are forced to use the tabview control from the xct. Which kind of limits what's we could have achieved if it was done using shell..

**Comment by nextcodelab** [comment]
For me I used BlazorWebView for some of my Pages and native NavigationPage since .net maui Shell is not complete yet. few controls bugs and not fully customizable. I also used native progress indicator and really feel native like dialogs.
Shell can't do this.

![Screenshot_20220515-211122](https://user-images.githubusercontent.com/73583588/168475058-8235f30b-1135-4526-97b7-2d985fa55d73.jpg)

```xaml
<Grid>
        <b:BlazorWebView x:Name="rzor" HostPage="wwwroot/index.html">

            <b:BlazorWebView.RootComponents>
                <b:RootComponent ComponentType="{x:Type local:HighLightsPage}" Selector="#app" />
            </b:BlazorWebView.RootComponents>
        </b:BlazorWebView>
        <ActivityIndicator
            x:Name="activityIndicator"
            Background="transparent"
            BackgroundColor="Transparent"
            HeightRequest="60"
            HorizontalOptions="Center"
            IsRunning="True"
            IsVisible="true"
            VerticalOptions="Center"
            WidthRequest="60"
            Color="BlueViolet" />
    </Grid>
```
```cs
public HighLightsXamlPage()
	{
		InitializeComponent();
		NavigationPage.SetHasNavigationBar(this, false);
		Current = this;
		activityIndicator.Color = App.AccentColor;
	}
	public static HighLightsXamlPage Current { get; private set; }
	public void GoBack()
	{
	   MainPage.Current.GoBack(this);
	}
    public void ShowIndicator()
    {
        if (activityIndicator == null)
            return;
        activityIndicator.IsRunning = true;
        activityIndicator.IsVisible = true;
    }
    public void HideIndicator()
    {
        if (activityIndicator == null)
            return;
        activityIndicator.IsRunning = false;
        activityIndicator.IsVisible = false;
    }
```
